lib/api/core/plugins/pluginsManager.js

```diff
@@ -312,34 +345,124 @@ class PluginsManager {
-        instances: plugin.config.threads,
-        killTimeout: plugin.config.killTimeout || 6000,
-        maxMemoryRestart: plugin.config.maxMemoryRestart || '1G',
-        watch: false
-      });
-    })
-    .catch(err => {
-      if (err) {
-        return Bluebird.reject(new PluginImplementationError('Error while starting worker plugin: '.concat(pluginName)));
+  const pluginList = fs.readdirSync(pluginsDir)
+    .filter(element => {
+      const pluginPath = path.join(pluginsDir, element);
+
+      try {
+        const elStat = fs.statSync(pluginPath);
+        return elStat.isDirectory();
+      }
+      catch(e) {
+        throw new PluginImplementationError(`Unable to load plugin from path "${pluginPath}"; this may occurs with dead symbolic links; ${e.message}`); // eslint-disable-line no-console
       }
     });
+
+  debug('loading plugins: %a', pluginList);
+
+  pluginList.forEach(pluginDirName => {
+    const
+      pluginPath = path.resolve(pluginsDir, pluginDirName),
+      plugin = {
+        name: path.basename(pluginPath), // we use only basename of plugin path to allow multi instances of same plugin
+        path: pluginPath,
+        object: null,
+        config: {},
+        manifest: Object.assign({}, DefaultManifest)
+      };
+
+    debug('[%s] loading plugin from directory "%s"', plugin.name, plugin.path);
+
+    try {
+      plugin.manifest = Object.assign(plugin.manifest, require(path.resolve(plugin.path, 'manifest.json')));
+    }
+    catch(e) {
+      // do nothing, simply use default manifest
+    }
+
+    // check for kuzzle core version prerequisite
+    if (!semver.satisfies(kuzzleVersion, plugin.manifest.kuzzleVersion)) {
+      throw new PluginImplementationError(`Unable to validate plugin "${plugin.name}"; required kuzzle version (${plugin.manifest.kuzzleVersion}) does not satisfies current: ${kuzzleVersion}`);
+    }
+
+    // check for existing custom configuration
+    plugin.config = config[plugin.name] || plugin.config;
+
+    // load plugin object
+    try {
+      const PluginClass = require(plugin.path);
+      plugin.object = new PluginClass();
+    } catch (e) {
+      throw new PluginImplementationError(`Unable to require plugin "${plugin.name}" from directory "${plugin.path}"; ${e.message}`);
+    }
+
+    // check plugin worker prerequisites
+    if (plugin.config.threads > 0) {
+      if (plugin.manifest.threadable) {
+        if (plugin.object.pipes || plugin.object.controllers || plugin.object.routes || plugin.object.strategies) {
+          throw new PluginImplementationError(`The plugin "${plugin.name}" is configured to run as worker but this plugin register non threadable features (pipes, controllers, routes, strategies)`);
+        }
+      }
+      else {
+        throw new PluginImplementationError(`The plugin "${plugin.name}" is configured to run as worker but this plugin does not support multi-threading; check "threadable" property in manifest.json plugin file`);
```

Comment:
Instead of pointing to the `manifest.json` file, suggesting that it can be changed to force it to be threadable against the will of plugin developers, I would instead put this kind of message:

`The plugin "${plugin.name}" is configured to run as a worker, but this feature has been explicitly disabled by their authors`